packages/wrangler/src/__tests__/tail.test.ts

```diff
@@ -67,38 +65,101 @@ describe("tail", () => {
 			expect(api.requests.creation.length).toStrictEqual(1);
 			expect(api.requests.deletion.count).toStrictEqual(0);
 
-			api.ws.close();
+			await api.closeHelper();
 			expect(api.requests.deletion.count).toStrictEqual(1);
 		});
 		it("should connect to the worker assigned to a given route", async () => {
 			const api = mockWebsocketAPIs();
 			expect(api.requests.creation.length).toStrictEqual(0);
 
-			mockGetZoneFromHostRequest("example.com", "test-zone");
-			mockCollectKnownRoutesRequest([
-				{
-					pattern: "example.com/*",
-					script: "test-worker",
-				},
-			]);
+			msw.use(
+				rest.get(`*/zones`, (req, res, ctx) => {
+					expect(req.url.searchParams.get("name")).toBe("example.com");
+					return res.once(
+						ctx.status(200),
+						ctx.json({
+							success: true,
+							errors: [],
+							messages: [],
+							result: [{ id: "test-zone" }],
+						})
+					);
+				})
+			);
+			msw.use(
+				rest.get(`*/zones/:zoneId/workers/routes`, (req, res, ctx) => {
+					return res.once(
+						ctx.status(200),
+						ctx.json({
+							success: true,
+							errors: [],
+							messages: [],
+							result: [
+								{
+									pattern: "example.com/*",
+									script: "test-worker",
+								},
+							],
+						})
+					);
+				})
+			);
```

Comment:
I think this would be slightly cleaner if it was a reimplementation of `mockGetZoneFromHostRequest` and `mockCollectKnownRoutesRequest`, since this adds a lot of boilerplate to each test

Reply:
I will do that with a future PR when other files using those abstractions/helpers are also converted so there is no conflict with the custom mocks and MSW; they conflict when used in the same test file.